src/System.Private.CoreLib/src/System/Reflection/Runtime/Assemblies/AssemblyNameHelpers.cs

```diff
@@ -32,8 +32,18 @@ public static String ComputeDisplayName(RuntimeAssemblyName a)
 
             if (a.Version != null)
             {
-                sb.Append(", Version=");
-                sb.Append(a.Version.ToString());
+                Version canonicalizedVersion = a.Version.CanonicalizeVersion();
+                if (canonicalizedVersion.Major != ushort.MaxValue)
```

Comment:
Unfortunately, it's already hard-coded in various places like CoreCLR and Version.cs (in the shared place) and as Int32 in some places and UInt16 in others, so a named constant is something that would be best considered holistically.